tt_metal/distributed/mesh_device.cpp

```diff
@@ -182,28 +210,77 @@ std::vector<chip_id_t> SystemMesh::Impl::get_mapped_physical_device_ids(const Me
                 physical_coordinate,
                 physical_device_id);
         }
+        return physical_device_ids;
+    } 
+    bool requires_rotation = requested_num_rows > system_mesh_rows || requested_num_cols > system_mesh_cols;
+    bool can_rotate = requested_num_rows <= system_mesh_cols && requested_num_cols <= system_mesh_rows;
+    bool should_rotate = requires_rotation && can_rotate && !is_single_row_or_column;
```

Comment:
`is_single_row_or_column` here is guaranteed to be false, right?

Reply:
Good catch -- I've cleaned up some of the conditions and handling.